### PR TITLE
[DOCS] Expands inference API docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR: 
* adds an admonition about the difference between the inference API and the ML inference function,
* highlights that inference can use OpenAI,
* highlights when to specify `sparse_embedding` and `text_embedding`.

### Preview

[Inference APIs](https://elasticsearch_104047.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/inference-apis.html)
[PUT Inference API](https://elasticsearch_104047.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-inference-api.html)

### CODE REVIEW

File: docs/reference/inference/put-inference.asciidoc [hunk 1]
```diff
@@ -27,6 +32,10 @@ Creates a model to perform an {infer} task.
 The create {infer} API enables you to create and configure an {infer} model to
 perform a specific {infer} task.
 
+The following services are available through the {infer} API:
+* ELSER
+* OpenAI
+
```
Comment (davidkyle) [inline docs/reference/inference/put-inference.asciidoc:38]:
```suggestion
* Hugging Face
```

File: docs/reference/inference/put-inference.asciidoc [hunk 2]
```diff
@@ -52,8 +61,8 @@ The type of the {infer} task that the model will perform. Available task types:
 (Required, string)
 The type of service supported for the specified task type.
 Available services:
-* `elser`,
-* `openai`.
+* `elser`: specify the `sparse_embedding` task type to use the ELSER service.
+* `openai`: specify the `text_embedding` task type to use the OpenAI service.
```
Comment (davidkyle) [inline docs/reference/inference/put-inference.asciidoc:65]:
```suggestion
* `openai`: specify the `text_embedding` task type to use the OpenAI service.
* `hugging_face`: specify the `text_embedding` task type to use the Hugging Face service.
```

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by szabosteve** [comment]
@elasticsearchmachine test this please

**Pull request merged**